specs/casper_sharding_v2.1.md

```diff
@@ -581,27 +581,43 @@ Finally:
-
-* Logic for the formats of shard chains, who proposes shard blocks, etc. (in an initial release, if desired we could make crosslinks just be Merkle roots of blobs of data; in any case, one can philosophically view the whole point of the shard chains as being a coordination device for choosing what blobs of data to propose as crosslinks)
-* Logic for inducting queued validators from the PoW chain
-* Penalties for signing or attesting to non-canonical-chain blocks (update: may not be necessary, see https://ethresear.ch/t/attestation-committee-based-full-pos-chains/2259)
-* Per-validator proofs of custody, and associated slashing conditions
-* Versioning and upgrades
-
-Slashing conditions may include:
-
-
-    Casper FFG slot equivocation [done]
-    Casper FFG surround [done]
-    Beacon chain proposal equivocation [done]
-    Shard chain proposal equivocation
-    Proof of custody secret leak
-    Proof of custody wrong custody bit
-    Proof of custody no secret reveal
-    RANDAO leak
+### TODO
+
+Note: This spec is ~60% complete.
+
+* [ ] Specify how `crystallized_state_root` and `active_state_root` are constructed
+* [ ] Specify the shard chain blocks, blobs, proposers, etc.
+* [ ] Specify the rules for forced deregistrations
+* [ ] Add a double-batched Merkle accumulator for historical beacon chain blocks
+* [ ] Specify the various assumptions (global clock, validator honesty, validator liveness, etc.)
+* [ ] Specify in Solidity the registration contract on the PoW chain
+* [ ] Specify the RANDAO logic, including the hardening against orphaned reveals
+* [ ] Add per-validator proofs of custody, including slashing conditions
+* [ ] Use a separate hash function for the proof of possession
+* [ ] Add an appendix for BLS12-381
+* [ ] Add an appendix on gossip networks and the offchain signature aggregation logic
+* [ ] Comprehensively and clearly define all the terms in the glossary
+* [ ] Clearly document the various edge cases, e.g. with committee sizing
+* [ ] Rework the document for readability
+* [ ] Undergo peer review, security audits and formal verification
+
+**Possible changes and additions**
+
+* [ ] Deprecate Wei and use 64-bit balances, consistent with a 64-bit EVM2.0
+* [ ] Allow for deposits larger than 32 ETH, as well as deposit top ups
+* [ ] Add penalties for a deposit below 32 ETH (or some other threshold)
+* [ ] Add a `SpecialObject` to change the `withdrawal_shard_id`, `withdrawal_address` or `randao_commitment`
+* [ ] Replace Blake with a STARK-friendly hash function
+* [ ] Replace the IMD fork choice rule with LMD
+* [ ] Merge `crystallized_state_root` and `active_state_root` into a single root
+* [ ] Add Merklelisation of the state root(s) for light clients
+* [ ] Add logic for versioning and upgrades
+* [ ] Get rid of dynasties
+* [ ] Add a RANDAO slashing condition for early leakage
+* [ ] Rework the `ShardAndCommittee` data structures
+* [ ] Reduce the slot duration to 8 seconds
```

Comment:
Does it depend on benchmarks?

Reply:
Partly. We can also make design changes to safely allow for faster slots. E.g. "Allow for the delayed inclusion of aggregated signatures"